### PR TITLE
[GLM10B] Fix rouge calculation 

### PR DESCRIPTION
**Pull request by LemonNoel**
### PR types

Bug fixes

### PR changes

Others

### Description

- The prediction tokens were selected by `attention_mask` in previous version, and `padding` tokens were not ignored when calculate rouge scores, which leads to a lower metric number.

### CODE REVIEW

**Review by wawltor — APPROVED** [review]
LGTM
